### PR TITLE
Fixes #7718: Put a makefile inside each directory of rudder-packages

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7718

### CODE REVIEW

**Comment by peckpeck** [comment]
We keep the name package.makefile because it reminds us its origin and not Makefile because it is not meant to be run from the root path.
The file is a direct copy from packaging branch 2.11

File: packages.makefile [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+include $(TOOLS_DIR)/build-package.makefile
+
+ARCHI := $(shell uname -m)
+OS := $(shell . SOURCES/rudder-sources/rudder-packages/detect_os.sh && echo $${OS})
+OSSP := $(shell . SOURCES/rudder-sources/rudder-packages/detect_os.sh && echo $${OSSP})
+OSVERSION := $(shell . SOURCES/rudder-sources/rudder-packages/detect_os.sh && echo $${OSVERSION})
```
Comment (jooooooon) [inline packages.makefile]: Shouldn't we adapt this path to use the local copy of detect_os.sh ?

File: packages.makefile [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+include $(TOOLS_DIR)/build-package.makefile
```
Comment (jooooooon) [inline packages.makefile]: Why not include this file too?

Comment (jooooooon) [inline packages.makefile]: By "include" I mean "copy it to rudder-packages"

File: rudder-policy-templates/Makefile [hunk 3]
```diff
@@ -0,0 +1 @@
+include ../packages.makefile
```
Comment (jooooooon) [inline rudder-policy-templates/Makefile]: Wow. This package should NOT still be around - it was deprecated in Rudder 2.4!

Comment (jooooooon) [inline rudder-policy-templates/Makefile]: (not to worry for this PR, of course, but we should remove it separately)

File: packages.makefile (hunk 2 above)
Comment (peckpeck) [inline packages.makefile]: I don't know where TOOL directory is nor how the file is copied there.
So I leaved everything as it was before.

File: packages.makefile (hunk 1 above)
Comment (peckpeck) [inline packages.makefile]: This path will soon be a local copy, cf #851

File: rudder-policy-templates/Makefile (hunk 3 above)
Comment (jooooooon) [inline rudder-policy-templates/Makefile]: I've created https://www.rudder-project.org/redmine/issues/7723 for this

File: packages.makefile (hunk 1 above)
Comment (jooooooon) [inline packages.makefile]: As agreed, we'll make sure to change the path here to ../detect_os.sh, once we have that file copied to all our builders.

File: packages.makefile (hunk 2 above)
Comment (peckpeck) [inline packages.makefile]: Some of its content is Normation specific.
I think I will merge it back progressively directly into this file

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
The patch to copy detect_os .sh and friends has been merged so this can be merged too

File: rudder-policy-templates/Makefile (hunk 3 above)
Comment (jooooooon) [inline rudder-policy-templates/Makefile]: Since we've removed this package, you should remove this Makefile from your PR.

**Comment by peckpeck** [comment]
PR updated

**Comment by jooooooon** [comment]
I have finally finished reviewing this. My apologies for the very long delay on this one!

I think this is fine, however I have noticed that there are some changes to this Makefile in branches 3.0 (change handling and versions of JDK) and 3.2 (addition of Solaris support). We will need extra Pull Requests to add these changes back in, for those two branches.

Merging this as is for now.

**Pull request merged**